lizmap/modules/lizmap/classes/qgisForm.class.php

```diff
@@ -1173,7 +1176,7 @@ private function fillControlFromValueRelationLayer($fieldName, $formControl)
 
             // Add default empty value for required fields
             // Jelix does not do it, but we think it is better this way to avoid unwanted set values
-            if ($formControl->ctrl->required) {
+            if ($formControl->ctrl->required or $formControl->valueRelationData['allowNull']) {
```

Comment:
If the field is not required but the `allowNull` is checked the relational value list will not have an empty value.
I add `or $formControl->valueRelationData['allowNull']` in the `if` to add empty value when `allowNull` is checked .